### PR TITLE
Stockfish update: 16 -> 16.1

### PR DESCRIPTION
**Pull request by ornicar**
## Description of changes

Upgrades stockfish to 16.1. It now uses 2 NNUE files:
https://stockfishchess.org/blog/2024/stockfish-16-1/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/stockfish/default.nix [hunk 1]
```diff
@@ -11,29 +11,38 @@ let
...
     rev = "sf_${version}";
-    sha256 = "sha256-ASy2vIP94lnSKgxixK1GoC84yAysaJpxeyuggV4MrP4=";
+    sha256 = "sha256-xTtjfJgEHF0SQT9Fw/9RLZA0Quh00jrIbihr7IYCm2U=";
   };
 
   postUnpack = ''
     sourceRoot+=/src
-    echo ${nnue}
-    cp "${nnue}" "$sourceRoot/${nnueFile}"
+    echo ${nnueBig}
```
Comment (gador) [inline pkgs/games/stockfish/default.nix:42]: Is there a technical reason to `echo`?

Comment (ornicar) [reply pkgs/games/stockfish/default.nix:42]: Apparently not, I tried removing it and the package still builds.

The `echo` was [already there](https://github.com/NixOS/nixpkgs/pull/333634/files/b8f7c30f68e5bb6b80c1be10d2934b621ead8209..4cfb5179495a6d2484e62666cb7604b2f071d52e#diff-92bcc9bd77c656dfdd19309d9131d3deff552008583b4504eb4636ac53be54a1L35) and it felt out of scope to remove it.
But I can do it here if you like.

Comment (gador) [reply pkgs/games/stockfish/default.nix:42]: I can't think of a reason for keeping it. If the package builds and runs fine, I'd vote to remove it :+1:

Comment (ornicar) [reply pkgs/games/stockfish/default.nix]: done!

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 333634` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>stockfish</li>
<li>uchess</li>
</ul>
</details>

**Review by gador — APPROVED** [review]

**Pull request merged**